### PR TITLE
Refactoring kademliaid

### PR DESCRIPTION
**Pull request by maxluetkemeyer**
Somehow the bootstrapping does not work anymore

### CODE REVIEW

**Comment by maxluetkemeyer** [comment]
The official Go Documentation stated, that things such as an byte slice should be passed by value.
https://go.dev/doc/faq#methods_on_values_or_pointers

Bigger stuff (like contacts) should be passed by pointer.

**Comment by Kyrne** [comment]
My guess to why bootstrapping does not work is that it is getting blocked by empty channel. Cleanest fix might be to close and recreate the channel for each iteration. Haven't had time to look into the rest of the pr, will do after lunch :D

File: internal/kademlia/kademliaid/calc_distance.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package kademliaid
+
+import (
+	"fmt"
+)
+
+// CalcDistance returns a new instance of a KademliaID that is built
+// through a bitwise XOR operation between kademliaID and target
+func (kademliaID KademliaID) CalcDistance(target KademliaID) KademliaID {
+	byteSlice, _ := bitwiseXOR(kademliaID.Bytes(), target.Bytes())
```
Comment (Kyrne) [inline internal/kademlia/kademliaid/calc_distance.go:10]: should we handle this error?

**Review by Kyrne — COMMENTED** [review]
lgtm, I'll push a fix for the bootstrap

File: internal/kademlia/kademliaid/calc_distance.go (hunk 1 above)
Comment (maxluetkemeyer) [reply internal/kademlia/kademliaid/calc_distance.go:10]: I split the function for testing reasons.
Two kademlia ids always have the same length, so I think it cannot fail. 
What do u think?

Comment (Kyrne) [reply internal/kademlia/kademliaid/calc_distance.go:10]: Yeah, that's true. This should be fine then :D

**Comment by maxluetkemeyer** [comment]
Afterward u can merge the pr :)

**Pull request merged**